### PR TITLE
remove AuthorizeFolder in secure user data

### LINKED ISSUE #5452: Secure user data - remove AuthorizeFolder/AuthorizePage
1. Replace SSL with HTTPS.
1. Once the default authorization is set requiring the user to be authenticated, no longer need the following:

```
services.AddMvc();
        //.AddRazorPagesOptions(options =>
        //{
        //    options.Conventions.AuthorizeFolder("/Account/Manage");
        //    options.Conventions.AuthorizePage("/Account/Logout");
        //});
```

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #5452 
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authorization/secure-data?branch=pr-en-us-5451#require-authenticated-users)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
Here is the big security change:
image

**Comment by HaoK** [comment]
Looks good to me, does this look ok @blowdart @javiercn ?

File: aspnetcore/security/authorization/secure-data/samples/final2/Startup.cs [hunk 1]
```diff
@@ -38,28 +38,28 @@ public void ConfigureServices(IServiceCollection services)
...
             // requires using Microsoft.AspNetCore.Mvc;
             services.Configure<MvcOptions>(options =>
             {
-                // Set LocalTest:skipSSL to true to skip SSL requrement in 
+                // Set LocalTest:skipHTTPS to true to skip SSL requrement in 
                 // debug mode. This is useful when not using Visual Studio.
-                if (Environment.IsDevelopment() && !skipSSL)
+                if (Environment.IsDevelopment() && !skipHTTPS)
                 {
                     options.Filters.Add(new RequireHttpsAttribute());
```
Comment (javiercn) [inline aspnetcore/security/authorization/secure-data/samples/final2/Startup.cs]: We don't recommend using RequireHttpsAttribute and I'm going to suggest deprecating it.

Comment (Rick-Anderson) [reply aspnetcore/security/authorization/secure-data/samples/final2/Startup.cs]: @javiercn  Should I replace it with `AddRedirectToHttps`? Don't we need to enforce HTTPS?

**Comment by Rick-Anderson** [comment]
@javiercn see #5393 which tracks 2.1 Improvements to HTTPS

**Pull request merged**